runIde runs an IDE using JRE instead of JDK

Hi, I'm running `0.3.6` version of the plugin and encountered the following error:
```
17:12:07: Executing task 'runIde'...

> Task :compileKotlin UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :patchPluginXml UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :instrumentCode UP-TO-DATE
> Task :postInstrumentCode
> Task :inspectClassesForKotlinIC UP-TO-DATE
> Task :jar UP-TO-DATE
> Task :prepareSandbox UP-TO-DATE

> Task :runIde
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=250m; support was removed in 8.0

JDK Required: JDK classes seem to be not on IDEA classpath.
Please ensure you run the IDE on JDK rather than JRE.

> Task :runIde FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runIde'.
> Process 'command '/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/bin/java'' finished with non-zero exit value 4

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 10s
9 actionable tasks: 2 executed, 7 up-to-date
Process 'command '/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/bin/java'' finished with non-zero exit value 4
17:12:18: Task execution finished 'runIde'.
```

More details, when running `./gradlew runIde --info` command:
```
> Task :runIde
Task ':runIde' is not up-to-date because:
  Task.upToDateWhen is false.
Starting process 'command '/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/bin/java''. Working directory: /Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/bin Command: /Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/bin/java -Dapple.awt.fileDialogForDirectories=true -Dapple.laf.useScreenMenuBar=true -Didea.classpath.index.enabled=false -Didea.config.path=/Users/scana/work/private/ok-gradle/build/idea-sandbox/config -Didea.is.internal=true -Didea.platform.prefix=Idea -Didea.plugins.path=/Users/scana/work/private/ok-gradle/build/idea-sandbox/plugins -Didea.required.plugins.id=me.scana.okgradle -Didea.smooth.progress=false -Didea.system.path=/Users/scana/work/private/ok-gradle/build/idea-sandbox/system -XX:MaxPermSize=250m -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -ea -cp /Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/lib/tools.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/idea_rt.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/idea.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/bootstrap.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/extensions.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/util.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/openapi.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/trove4j.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/jdom.jar:/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/fcb66fea071aa9c4de3a607c505f61d29ac56d8e/ideaIC-LATEST-EAP-SNAPSHOT/lib/log4j.jar com.intellij.idea.Main
Successfully started process 'command '/Users/scana/.gradle/caches/modules-2/files-2.1/com.jetbrains/jbre/jbrex8u152b1248.8_osx_x64/jdk/Contents/Home/jre/bin/java''
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=250m; support was removed in 8.0

JDK Required: JDK classes seem to be not on IDEA classpath.
Please ensure you run the IDE on JDK rather than JRE.
```

Am I doing anything wrong? Is there any way to point the plugin to run the IDE from a different SDK?
If anything, I'm using 2018.2 version of Intellij IDEA.

I would be grateful for any tips on this matter.

--- a/src/main/groovy/org/jetbrains/intellij/Utils.groovy
+++ b/src/main/groovy/org/jetbrains/intellij/Utils.groovy
@@ -235,6 +235,15 @@ class Utils {
         }, recursively ? TrueFileFilter.INSTANCE : FalseFileFilter.FALSE)
     }
 
+    static String resolveToolsJar(String javaExec) {
+        String binDir = new File(javaExec).parent
+        if (OperatingSystem.current().isMacOsX()) {
+            return "$binDir/../../lib/tools.jar"
+        }
+        return "$binDir/../lib/tools.jar"
+    }
+
+
     static String getBuiltinJbreVersion(@NotNull File ideaDirectory) {
         def dependenciesFile = new File(ideaDirectory, "dependencies.txt")
         if (dependenciesFile.exists()) {

--- a/src/main/groovy/org/jetbrains/intellij/Utils.groovy
+++ b/src/main/groovy/org/jetbrains/intellij/Utils.groovy
@@ -243,7 +243,6 @@ class Utils {
         return "$binDir/../lib/tools.jar"
     }
 
-
     static String getBuiltinJbreVersion(@NotNull File ideaDirectory) {
         def dependenciesFile = new File(ideaDirectory, "dependencies.txt")
         if (dependenciesFile.exists()) {

--- a/src/main/groovy/org/jetbrains/intellij/tasks/RunIdeTask.groovy
+++ b/src/main/groovy/org/jetbrains/intellij/tasks/RunIdeTask.groovy
@@ -1,14 +1,10 @@
 package org.jetbrains.intellij.tasks
 
-import org.gradle.api.GradleException
+
 import org.gradle.api.tasks.*
 import org.gradle.internal.os.OperatingSystem
 import org.gradle.util.CollectionUtils
 import org.jetbrains.intellij.Utils
-
-import java.nio.file.Files
-import java.nio.file.Path
-import java.nio.file.Paths
 
 class RunIdeTask extends JavaExec {
     private static final def PREFIXES = [IU: null,
@@ -133,7 +129,7 @@ class RunIdeTask extends JavaExec {
     private void configureClasspath() {
         File ideaDirectory = getIdeaDirectory()
         def executable = getExecutable()
-        if (executable) classpath += project.files(resolveToolsJar(executable))
+        if (executable) classpath += project.files(Utils.resolveToolsJar(executable))
         classpath += project.files("$ideaDirectory/lib/idea_rt.jar",
                 "$ideaDirectory/lib/idea.jar",
                 "$ideaDirectory/lib/bootstrap.jar",
@@ -143,23 +139,6 @@ class RunIdeTask extends JavaExec {
                 "$ideaDirectory/lib/trove4j.jar",
                 "$ideaDirectory/lib/jdom.jar",
                 "$ideaDirectory/lib/log4j.jar")
-    }
-
-    private String resolveToolsJar(String javaExec) {
-        String binDir = new File(javaExec).parent
-        // tools.jar for windows & linux
-        // sibling of bin dir
-        Path toolsJar = Paths.get("$binDir/../lib/tools.jar")
-        // tools.jar for osx
-        // sibling of the parent of bind dir
-        Path altToolsJar = Paths.get("$binDir/../../lib/tools.jar")
-        if (Files.exists(toolsJar)) {
-            return toolsJar.toString()
-        } else if (Files.exists(altToolsJar)) {
-            return altToolsJar.toString()
-        }
-
-        throw new GradleException("Unable to resolve tools.jar for $javaExec.")
     }
 
     def configureSystemProperties() {

--- a/src/main/groovy/org/jetbrains/intellij/tasks/RunIdeTask.groovy
+++ b/src/main/groovy/org/jetbrains/intellij/tasks/RunIdeTask.groovy
@@ -1,10 +1,14 @@
 package org.jetbrains.intellij.tasks
 
+import org.gradle.api.GradleException
 import org.gradle.api.tasks.*
-import org.gradle.internal.jvm.Jvm
 import org.gradle.internal.os.OperatingSystem
 import org.gradle.util.CollectionUtils
 import org.jetbrains.intellij.Utils
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
 
 class RunIdeTask extends JavaExec {
     private static final def PREFIXES = [IU: null,
@@ -129,7 +133,7 @@ class RunIdeTask extends JavaExec {
     private void configureClasspath() {
         File ideaDirectory = getIdeaDirectory()
         def executable = getExecutable()
-        if (executable) classpath += project.files("${new File(executable).parent}/../lib/tools.jar")
+        if (executable) classpath += project.files(resolveToolsJar(executable))
         classpath += project.files("$ideaDirectory/lib/idea_rt.jar",
                 "$ideaDirectory/lib/idea.jar",
                 "$ideaDirectory/lib/bootstrap.jar",
@@ -139,6 +143,23 @@ class RunIdeTask extends JavaExec {
                 "$ideaDirectory/lib/trove4j.jar",
                 "$ideaDirectory/lib/jdom.jar",
                 "$ideaDirectory/lib/log4j.jar")
+    }
+
+    private String resolveToolsJar(String javaExec) {
+        String binDir = new File(javaExec).parent
+        // tools.jar for windows & linux
+        // sibling of bin dir
+        Path toolsJar = Paths.get("$binDir/../lib/tools.jar")
+        // tools.jar for osx
+        // sibling of the parent of bind dir
+        Path altToolsJar = Paths.get("$binDir/../../lib/tools.jar")
+        if (Files.exists(toolsJar)) {
+            return toolsJar.toString()
+        } else if (Files.exists(altToolsJar)) {
+            return altToolsJar.toString()
+        }
+
+        throw new GradleException("Unable to resolve tools.jar for $javaExec.")
     }
 
     def configureSystemProperties() {

